src/controllers/bookController.js

```diff
@@ -13,39 +13,46 @@ class BookController {
     this.formView.setOnSubmit(this.handleFormSubmit.bind(this));
     this.listView.setOnEdit(this.handleEdit.bind(this));
     this.listView.setOnDelete(this.handleDelete.bind(this));
+
+    this.createBookButton = document.querySelector('.create-book');
+    this.createBookButton.addEventListener(
+      'click',
+      this.handleCreateBookButtonClick.bind(this),
+    );
+  }
+
+  handleCreateBookButtonClick() {
+    this.formView.render();
   }
 
-  async handleFormSubmit(title, author, genre, publishedYear) {
+  showSuccessMessage(message) {
+    // Show the success message using the toast
+    this.toast.show(message, 'success');
+  }
+
+  async handleFormSubmit(title, author, category, publishedYear) {
     try {
       if (this.model.currentBook) {
-        await this.updateBook(title, author, genre, publishedYear);
+        await this.updateBook(title, author, category, publishedYear);
```

Comment:
Im not sure about this, you wanna update but do you not need an id?

Reply:
i will get id on bookModel and in controller Im call `this.model.curentBook` and update.
```
  updateBook(updatedBook) {
    const index = this.books.findIndex((book) => book.id === updatedBook.id);
    if (index !== -1) {
      const { id, title, author, category, publishedYear } = updatedBook;
      this.books[index] = new Book(id, title, author, category, publishedYear);
    }
  }
```
this is updateBook on bookmodel